napari/utils/migrations.py

```diff
@@ -218,3 +245,71 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class DeprecatingDict(dict[str, Any]):
```

Comment:
I'm not sure if this should be part of a public API. 
At least I will put in docstring, that is not intended to be used outside napari as may be mutated. 

But it is not a strong opinion.

Reply:
I'm fine adding a leading `_` and will add to the docstring to clarify that the user should use this as as if it were a `dict[str, Any]`.

Reply:
An instance of this will be publicly accessible though.